sonar/modules/documents/tasks.py

```diff
@@ -22,7 +22,18 @@
 from invenio_db import db
 from invenio_indexer.api import RecordIndexer
 
-from sonar.modules.documents.api import DocumentRecord
+from sonar.modules.documents.urn import Urn
+
+
+@shared_task(ignore_result=True)
+def register_urn_code_from_document(record):
+    """Register the urn pid for a given document.
+
+    :param record records_to_import: the document.
+    """
+    from sonar.modules.documents.api import DocumentRecord
+    if DocumentRecord.get_urn_codes(record):
+        Urn.register_urn_code_from_document(record)
 
```

Comment:
Nice to have a return value. This return value could be used for debugging in flower later.